src/realm/array_object_id.cpp

```diff
@@ -99,13 +99,24 @@ void ArrayObjectId::move(ArrayObjectId& dst_arr, size_t ndx)
 
     const auto n_to_move = old_src_size - ndx;
 
+    dst_arr.copy_on_write();
+    dst_arr.ensure_minimum_width(this->m_ubound);
```

Comment:
Does this make sense in this context?

Reply:
You're right, this shouldn't be necessary here, I'll remove it.